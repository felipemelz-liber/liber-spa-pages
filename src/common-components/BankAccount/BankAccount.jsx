import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { Container, ImagePlaceHolder, ImageLogo, InfoContainer, Title } from './BankAccount.styles';
import defaultImage from '../../../../assets/images/bank-logos/default.png';

const formatDigit = digit => (digit && digit !== '' ? `-${digit}` : '');

export const BankAccount = ({ bankCode, bankName, branch, branchCd, account, accountCd }) => {
  const [image, setImage] = useState(null);

  useEffect(() => {
    import(`../../../../assets/images/bank-logos/banco_0${bankCode}.png`)
      .then(res => {
        setImage(res.default);
      })
      .catch(() => {
        setImage(defaultImage);
      });
  }, []);

  return (
    <Container>
      {image ? <ImageLogo src={image} alt={`Logo do ${bankName}`} /> : <ImagePlaceHolder />}
      <InfoContainer>
        <Title>{`${bankCode} - ${bankName}`}</Title>
        <div>{`AgÃªncia ${branch}${formatDigit(branchCd)}`}</div>
        <div>{`Conta ${account}${formatDigit(accountCd)}`}</div>
      </InfoContainer>
    </Container>
  );
};

BankAccount.propTypes = {
  branch: PropTypes.string,
  branchCd: PropTypes.string,
  account: PropTypes.string,
  accountCd: PropTypes.string,
  bankCode: PropTypes.string,
  bankName: PropTypes.string,
};

BankAccount.defaultProps = {
  branch: '',
  branchCd: '',
  account: '',
  accountCd: '',
  bankCode: '',
  bankName: '',
};

export default BankAccount;
