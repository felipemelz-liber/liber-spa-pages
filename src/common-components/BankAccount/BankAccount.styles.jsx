import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: row;
`;

export const ImagePlaceHolder = styled.div`
  width: 120px;
  height: 120px;
  border-radius: 4px;
  border: solid 1px #cdd7df;
  padding: 1px;
`;

export const ImageLogo = styled.img`
  width: 120px;
  height: 120px;
  border-radius: 4px;
  border: solid 1px #cdd7df;
  padding: 1px;
`;

export const InfoContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 16px;

  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
`;

export const Title = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
  margin-bottom: 8px;
  max-width: 408px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;
