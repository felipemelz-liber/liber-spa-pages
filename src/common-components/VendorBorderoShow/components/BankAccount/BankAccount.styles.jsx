import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  padding: 2px 0px;
`;

export const ImageContainer = styled.div`
  width: 120px;
  height: 120px;
  object-fit: contain;
  border-radius: 4px;
  border: solid 1px #cdd7df;
  background-color: #f7f7f7;
  margin-right: 16px;
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Title = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
  margin-bottom: 8px;
  margin-top: 16px;
`;

export const Description = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  font-stretch: normal;
  font-style: normal;
  line-height: 1.38;
  letter-spacing: 0.06px;
  text-align: left;
  color: #6287a7;
`;
