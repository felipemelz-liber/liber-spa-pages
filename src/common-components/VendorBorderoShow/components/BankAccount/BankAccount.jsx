import React from 'react';
import PropTypes from 'prop-types';
import { Container } from './BankAccount.styles';
import { BankAccount as BankAccountComponent } from '../../../BankAccount/BankAccount';

const BankAccount = props => {
  return (
    <Container>
      <BankAccountComponent {...props} />
    </Container>
  );
};

BankAccount.propTypes = {
  branch: PropTypes.string,
  branchCd: PropTypes.string,
  account: PropTypes.string,
  accountCd: PropTypes.string,
  bankName: PropTypes.string,
  bankCode: PropTypes.string,
};

BankAccount.defaultProps = {
  branch: '',
  branchCd: '',
  account: '',
  accountCd: '',
  bankName: '',
  bankCode: '',
};

export default BankAccount;
