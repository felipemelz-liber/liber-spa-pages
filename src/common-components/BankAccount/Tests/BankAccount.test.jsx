import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import BankAccount from '../BankAccount';

describe('BankAccount component tests', () => {
  configure({ adapter: new Adapter() });

  const props = {
    branch: '001',
    branchCd: '1',
    account: '5454',
    accountCd: '1',
    bankCode: '001',
    bankName: 'Banco do Brasil S.A.',
  };

  it('should render correctly without props', () => {
    const wrapper = shallow(<BankAccount />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const wrapper = shallow(<BankAccount {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
