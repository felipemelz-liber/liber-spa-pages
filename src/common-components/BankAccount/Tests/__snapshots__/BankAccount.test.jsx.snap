// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankAccount component tests should render correctly with props 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <styled.div>
      001 - Banco do Brasil S.A.
    </styled.div>
    <div>
      Agência 001-1
    </div>
    <div>
      Conta 5454-1
    </div>
  </styled.div>
</styled.div>
`;

exports[`BankAccount component tests should render correctly without props 1`] = `
<styled.div>
  <styled.div />
  <styled.div>
    <styled.div>
       - 
    </styled.div>
    <div>
      Agência 
    </div>
    <div>
      Conta 
    </div>
  </styled.div>
</styled.div>
`;
